DBConnector.class.php

```diff
@@ -125,6 +125,19 @@ public function ipAddress()
         return $ip;
     }
 
+    public function checkPermission()
+    {
+        $ip = $this->ipAddress();
+        $ip_list = array();
```

Comment:
The array `$ip_list` must also contain "127.0.0.1", because PalMA sends control commands to itself, for example when a VNC connection starts.

The latest release is broken because it blocks such requests.

Reply:
While we are at it: are the IPs all IPv4 addresses or would we have to take IPv6 into account as well? At least the local connections could also come from ::1 I think...

Reply:
I think it depends on the IP which was assigned to the name of the host in /etc/hosts.

Reply:
A better solution might avoid such local connections.

Reply:
b69f4421484424ded248c55cba231adf819e26cc should fix this.

Reply:
It does, thank you. While we are at fixing known security problems, we should also use the new code to block VNC connections from unregistered users.